pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -655,12 +655,56 @@ protected void internalDeletePartitionedTopic(AsyncResponse asyncResponse, boole
         });
     }
 
-    protected void internalUnloadTopic(boolean authoritative) {
+    protected void internalUnloadTopic(AsyncResponse asyncResponse, boolean authoritative) {
         log.info("[{}] Unloading topic {}", clientAppId(), topicName);
         if (topicName.isGlobal()) {
             validateGlobalNamespaceOwnership(namespaceName);
         }
-        unloadTopic(topicName, authoritative);
+
+        PartitionedTopicMetadata partitionMetadata = getPartitionedTopicMetadata(topicName, authoritative, false);
```

Comment:
`AsyncResponse` indicates that this method is an async operation. So can we implement this method in an async way?

Reply:
Does it seem that my comment here was not addressed?

Reply:
implement getPartitionedTopicMetadataAsync in an async way?

Reply:
Yes. we need to make sure this method is also implemented in an async way.